src/org/traccar/notification/NotificationFormatter.java

```diff
@@ -68,31 +73,52 @@ public static VelocityContext prepareContext(long userId, Event event, Position
     }
 
     public static Template getTemplate(Event event, String path) {
+
+        String templateFilePath;
         Template template;
+
         try {
-            template = Context.getVelocityEngine().getTemplate(path + event.getType() + ".vm",
-                    StandardCharsets.UTF_8.name());
+            templateFilePath = Paths.get(path, event.getType() + ".vm").toString();
+            template = Context.getVelocityEngine().getTemplate(templateFilePath, StandardCharsets.UTF_8.name());
         } catch (ResourceNotFoundException error) {
             Log.warning(error);
-            template = Context.getVelocityEngine().getTemplate(path + "unknown.vm",
-                    StandardCharsets.UTF_8.name());
+            templateFilePath = Paths.get(path, "unknown.vm").toString();
+            template = Context.getVelocityEngine().getTemplate(templateFilePath, StandardCharsets.UTF_8.name());
         }
         return template;
     }
 
     public static MailMessage formatMailMessage(long userId, Event event, Position position) {
+        String templatePath = Context.getConfig().getString("mail.templatesPath", "mail");
         VelocityContext velocityContext = prepareContext(userId, event, position);
-        StringWriter writer = new StringWriter();
-        getTemplate(event, Context.getConfig().getString("mail.templatesPath", "mail") + "/")
-                .merge(velocityContext, writer);
-        return new MailMessage((String) velocityContext.get("subject"), writer.toString());
+        String formattedMessage = formatterLogic(velocityContext, userId, event, position, templatePath);
+
+        return new MailMessage((String) velocityContext.get("subject"), formattedMessage);
     }
 
     public static String formatSmsMessage(long userId, Event event, Position position) {
-        VelocityContext velocityContext = prepareContext(userId, event, position);
+        String templatePath = Context.getConfig().getString("sms.templatesPath", "sms");
+
+        return formatterLogic(null, userId, event, position, templatePath);
+    }
+
+    public static String formatForwarderMessage(Event event, Position position) {
```

Comment:
Why do we still have this and template files?

Reply:
Removed, done